Refac user login

miltonbsn opened the pull request:
# Summary 

Since we are upgrading most libs, including express and node.js itself, I am proposing a major refactoring to follow the good practices introduced since I developed the first version of these features. Besides apply new es6 standard code, I am changing significantly the core architecture on backend, following a domain driven design standard. The main goal is to allow new features to be easily introduced and tested

## List of changes

- Separates the coupling between app and server ;
- Complete rewrite of user endpoints, introducing a new package called user with all it's responsibilities; 
- Add tests with Jest, supertest and mockingoose;

miltonbsn committed:
Refac user login

miltonbsn committed:
Refac user login

miltonbsn committed:
Refac user create

miltonbsn committed:
Refac user create

miltonbsn committed:
Remove legacy userRouter.js

miltonbsn committed:
Remove legacy userRouter.js

miltonbsn committed:
Remove user.js from models folder

miltonbsn committed:
Remove user.js from models folder

miltonbsn committed:
Merge branch 'master' into refac-server-user

idmarjr committed:
Use Bootstrap in main.html table

idmarjr committed:
Style adjustment for table

- Visual adjustments;
- Show cursor pointer when :hover table rows (Thanks bootstrap for the hight specific selector that we need to overwrite :( )
-

idmarjr committed:
Table: Adjustments for .action-cell

- Remove depricated styles from structure.scss;
- Add .action-cell styles
   - Avoid absolute positioning action icons;
   - Add :hover interaction to action icons;
   - Enable pointer events to action icons only when :hover the icon itself;

idmarjr committed:
Remove unused button styles

idmarjr committed:
Code optimization for .br-button stylles

- Change unities from rem to px (for now);
- Remove some unecessary styles;
- Set .br-button as inline-block element;

idmarjr committed:
Migrade .contentHeader to bootstrap styles

- Use bootstrap classes like .container, .page-header, .pull-left, .pull-right;
- Remove non necessary clases like headerButton and .headerText;
- Remove styles related to the old markup;

idmarjr committed:
Remove reset.scss

Since we're moving to a more bootstrap like way, it make no sense to deal with one more layer of styles so I'm removing the original reset.css

Keep in mind it will break some things in the layout for a while.

- Define box-sizing directly in structure.scss
- Change default font-weight;
- Add cursor: pointer as default  for links.

idmarjr committed:
Visual fixes for jquery-nice-select elements

idmarjr committed:
Visual fix for few Bootstrap form elements

- .checkbox input vertical aligment;
- Use BR Modelo green for input:focus;

idmarjr committed:
main.html: Re-write app mainHeader using Bootstrap markup

idmarjr committed:
Small adjustment for buttons

idmarjr committed:
Move ModelProperty panel components to Bootstrap

- Rename classes from camelCase to dash-separated
    - .modelProperties --> .model-properties;
    - .propertiesContent --> .properties-content;
- Use bootstrap form markup to all for fields

idmarjr committed:
New conceptual workspace layout (Using Bootstrap)

- Move mainHeader to use Bootstrap markup
- Add new .content .page-header área to hold information/actions about the current model;
- Move toolbar from mainHeader to .tool-icons;
- Remove Save/Export actions from mainHeader
    - Save --> New .content .page-header
    - Export  to SQL--> New content .page-header .tool-icons;
- Remove Input to edit model name from mainHeader;
- Display current model name at the .content .page-header area;
- Size adjustments to all panels:
    - .stencil-list
   -  .model-area
    - .model-properties

idmarjr committed:
Remove unecessary styles from .view

idmarjr committed:
Adjustments for .feedback

- Move it to a new .workspace-area;
- Remove a tom of unecessary styles

idmarjr committed:
Add icon to "New model" button

- Add icon
- Adjust paddings to all .br-buttons;

idmarjr committed:
Add dummy for "Rename" option

Since the rename option was removed from the model workspace, I'm suggesting a option to edit the name of if in the list of all models

idmarjr committed:
Move CSS fixes of selec plugin to its relevant file

- Move from form.scss to selectOptions.scss;
- Adjust text aligment the rendered custom select;

idmarjr committed:
Fix hover state of .navbar-brand

feekosta commented on server_app/user/service.js:
```diff
@@ -0,0 +1,56 @@
+      });
+    
+      if(userDocument != null) {
+        resolve({
```
return in the if, to not execute the rest of the code

feekosta commented on server_app/user/service.js:
```diff
@@ -0,0 +1,56 @@
+      const user = await UserRepository.findOne({"login": mail});
+
+      if(user != null) {
+        reject({"code": "USER_ERROR_ALREADY_EXISTS"});
```
return in the if, to not execute the rest of the code

idmarjr committed:
Add .tool-icons .divider styles

idmarjr committed:
Remove unreferenced CSS

idmarjr committed:
Remove hacky solution to centralize modal vertically

idmarjr committed:
Remove unreferenced CSS

idmarjr committed:
Remove unreferenced CSS about .modalFeedback

idmarjr committed:
Remove unreferenced CSS

idmarjr committed:
Remove missused .addmodel class

- Remove multiple unecessary entries in the code;
- Move it's styles to .modal-content;

idmarjr committed:
.modal-title adjustments

- Use <h3> to leep things a bit more semantic;
- Use dash-separated class name;
- Remove font-size styles;

idmarjr committed:
Simplify DeleteModelModal code

- Remove unecessary markup;
- Keep code inline with bootstrap documentation;

idmarjr committed:
Make modals great again (And add few bugs in the way)

- Fix moda markup mess;
- Delete unecessary .modal-message class

idmarjr committed:
Remove .modal-input styles

idmarjr committed:
Better handle with inut form validation errors

- Replace outline with box-shadow effects;
- Deal with validation in .error  scenarios;

idmarjr committed:
Move styles related to form from modal.scss to form.scss

idmarjr committed:
Leave font-size global definition to Bootstrap

idmarjr committed:
Replace CSS comment header with Sass comment markup

Since we're here and we have Sass in the setup... WHy not?

idmarjr committed:
Add print option to .tool-icons menu

idmarjr committed:
Add option to generate logic model (Proposal)

idmarjr committed:
Save vertical pixels

- Reduce .workspace-header off space;
- Reduce padding of .tool-icons itens;
- Adjust model-properties dimentions to fit changes above;

idmarjr committed:
.paperScroller: Improve code legibility

idmarjr committed:
Remove unecessary box-sizing property

- We're already defining it to all elements

idmarjr committed:
Code cleanup at .papler-scroller

- Remove some unecessary declaratuins;
- Remove borwser prefix declarations (Those are handled by autoprefixer nowadays);

idmarjr committed:
Change .paper-scroller height definition

idmarjr committed:
Migrate markup of logic model view to Bootstrap

- Migrate general structure
- Still missing migrate forms linving inside of .model-properties;
- Add scope to show/hide .model-properties in logic view;
- Remove old CSS comment header;

idmarjr committed:
FInish migrate logic.html to bootstrap markup

- Use bootsrrap forms
- Remove a tom of unecessary markup/classes;

idmarjr committed:
Remove references of .specialElement

idmarjr committed:
Visual adjustments to

- Icon sizes adjustments;
- Adjustments for borders and paddings;

idmarjr committed:
Remove unecessary classes

idmarjr committed:
CSS not used anymore: kill ir with FIRE!

idmarjr committed:
Rewrite public pages

- Use part of Bootstrap markup;
- Remove a ton of out of date CSS

idmarjr committed:
Interface adjustments

idmarjr committed:
Remove outdated classes

idmarjr committed:
CSS cleanup

Remove a ton of out of date CSS

idmarjr committed:
Remove option to generate SQL from conceptual view

idmarjr committed:
Remove option to generatel logic model from logic view

idmarjr committed:
Small interface adjustments

miltonbsn committed:
Set up jest configuration

miltonbsn committed:
Set up jest configuration

miltonbsn committed:
Refac user handler

miltonbsn committed:
Refac user handler

miltonbsn committed:
Refac user service + test

miltonbsn committed:
Refac user service + test

miltonbsn committed:
Refac user validator + test

miltonbsn committed:
Refac user validator + test

idmarjr committed:
Enable print in logic view

idmarjr committed:
Adjust printing styles

idmarjr committed:
Interface adjustments

- .dropdown-tmp colors and positioning;
- Checkbox vertical aligment fix;

idmarjr committed:
.dropdown-tmp: Remove button outline when in focus

Not that good for acessibility but taking in consideration we have much bigger problems regarding this topic in the app I dont see as a problem right now But we need to address it in the future.)

idmarjr committed:
Add view for RenameModelModal

miltonbsn committed:
add supportive test packages

miltonbsn committed:
add supportive test packages

miltonbsn committed:
Separates server.js set up from app.js (good practice)

miltonbsn committed:
Separates server.js set up from app.js (good practice)

miltonbsn committed:
Testing user/handler.js

miltonbsn committed:
Testing user/handler.js

idmarjr committed:
FIx typo causing scss compilation to fail

idmarjr committed:
Remove jquery-nice-select as dependency

idmarjr committed:
Remove remaining jquery-nice-select references

- Remove from Gulp tasks;
- Remove CSS customizations;
- Remove links to js and CSS from views/index.html

idmarjr committed:
Update README.md

- Simplify language;
- Fix ordered lists;
- Remove gulp as dependency (Sin it's installed during setup);

idmarjr committed:
Merge pull request #102 from miltonbsn/bootstrap-markup

- Migrate markup to Bootstrap
- CSS code cleanup;
- Remove not used dependencies;
- Various interface optimizations;

idmarjr committed:
Update issue templates

rpuggian commented on server_app/user/service.js:
```diff
@@ -0,0 +1,56 @@
+    
+      if(userDocument != null) {
+        return resolve({
+          "sessionId": sessionId,
```
Miltão, What do you think about leaving the responsibility of setting up the user session to the handler? I noticed in the handler we already have the session ID. Maybe like that we don't need to verify != null here and always resolve the userDocument.

rpuggian commented on server_app/user/handler.js:
```diff
@@ -0,0 +1,62 @@
+      return res.status(422).send(validation.message);
+    }
+      
+    const userSession = await userService.login({username, password, sessionId});
```
Maybe you can build the user session here, and just find the user in the userService using the password encryption.

miltonbsn committed:
Deleting package-lock.json (Deploy heroku)

miltonbsn committed:
Apply pull request recomendations

miltonbsn committed:
Merge remote
